lib/mongodb.js

```diff
@@ -483,6 +482,121 @@ MongoDB.prototype.updateOrCreate = function updateOrCreate(model, data, options,
 };
 
 /**
+ * Parses the data input for replace operations and returns the
+ * sanitised version of the object.
+ *
+ * @param data
+ * @returns {*}
+ */
```

Comment:
Yes please, don't duplicate code if possible.

I am proposing to remove extended operators support from `replace`, as the intention is different here.